File: ScriptService/Dto/Workflows/WorkflowData.cs
using NightlyCode.Database.Entities.Attributes;

namespace ScriptService.Dto.Workflows {

    /// <summary>
    /// data of a workflow
    /// </summary>
    public class WorkflowData {

        /// <summary>
        /// name of workflow
        /// </summary>
        [Unique]
        public string Name { get; set; }

        /// <summary>
        /// default script language for expressions
        /// </summary>
        public ScriptLanguage? Language { get; set; }
    }
}

File: nc-scripts/src/app/dto/workflows/workflowdata.ts
import { ScriptLanguage } from "../scripts/scriptlanguage";

/**
 * basic data of a workflow
 */
export interface WorkflowData {
    
    /**
     * name of workflow
     */
    name: string,

    /**
     * default script language used for expressions
     */
    language?: ScriptLanguage
}